File: app/schemas/bestfit.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class BestFitCreate(BaseModel):
    bestfit_id: str
    user_id: str
    size_chart_id: str
    row_id: str
    product_name: str = Field(..., max_length=20)
    memo: Optional[str] = Field(None, max_length=100)
    favorite: bool = False
    pant_fit_type: Optional[str] = Field(None, examples=["wide", "straight", "slim"])


class BestFitResponse(BaseModel):
    bestfit_id: str
    user_id: str
    size_chart_id: str
    row_id: str
    product_name: str
    memo: Optional[str]
    favorite: bool
    pant_fit_type: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True

File: src/main/java/com/fitfounder/backend/domain/BestFit.java
package com.fitfounder.backend.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.time.OffsetDateTime;
import java.time.ZoneOffset;

@Entity
@Table(name = "bestfits")
@Getter
@Setter
public class BestFit {
    @Id
    @Column(name = "bestfit_id")
    private String bestfitId;

    @Column(name = "user_id", nullable = false)
    private String userId;

    @Column(name = "size_chart_id", nullable = false)
    private String sizeChartId;

    @Column(name = "row_id", nullable = false)
    private String rowId;

    @Column(name = "product_name", nullable = false, length = 20)
    private String productName;

    @Column(name = "memo", length = 100)
    private String memo;

    @Column(name = "favorite", nullable = false)
    private boolean favorite = false;

    @Column(name = "pant_fit_type")
    private String pantFitType;

    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;

    @PrePersist
    public void onCreate() {
        OffsetDateTime now = OffsetDateTime.now(ZoneOffset.UTC);
        createdAt = now;
        updatedAt = now;
    }

    @PreUpdate
    public void onUpdate() {
        updatedAt = OffsetDateTime.now(ZoneOffset.UTC);
    }
}
